### ISSUE TITLE
Unity Firebase DLLNotFoundException [Unity 2017.4]

### ISSUE THREAD

**Issue by deltamish** [firebase/quickstart-unity#226]
Hey,

After quite a lot of effort due to little difficult to understand documentation I finally manged to integrate Storage and RealTime database in our unity project. Everything is working fine in the Editor but in Android build its saying 
"DllNotFound Exception : FirebaseCppApp 5.3.1"

I am using internal Build system. 
Have SDK and NDK paths set properly. 
Using .Net 3.5 and .Net 2.0 (Full Not subset)

But still I am facing this issue. 
![2018-10-16-12-04-30](https://user-images.githubusercontent.com/9283808/46997034-3b6c5b00-d13c-11e8-9982-641ebc0d2873.png)

Thanks & Regards,

**Comment by SachinGanesh** [comment]
In my case the problem was caused because I used different versions of Firebase Messaging and Firebase Analytics. Problem was solved after updating all used sdks to same version.

**Comment by deltamish** [comment]
Hi,
The Issue was solved. 
1st Step
The Problem was actually Java. My environment variable had JAVA_HOME as "C:/Program Files/Java/jdk10.xx ;" And a semicolon at the end. So gradle could't read it. Once I removed it The error "JAVA_HOME is not set to valid JVM environment was gone" 

2nd Step
Android Studio/ Gradle doesn't support Java x64 and threw an error when opened manually. So Install JavaX86 version and set path to that and remember to remove semi colon in Environment variable if it has any, 

3rd Step
Ran Unity->Assets->Play service resolver->Android->Force resolve


**Bonus:**
Update Android SDK tools to 26.1.1 to solve **Manifest Merge Failed  Error**

Cheers
